src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/PersonBlockingRulesRepository.kt

```diff
@@ -0,0 +1,109 @@
+package uk.gov.justice.digital.hmpps.personrecord.jpa.repository
+
+import jakarta.persistence.EntityManager
+import jakarta.persistence.PersistenceContext
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
+import uk.gov.justice.digital.hmpps.personrecord.model.person.Person
+
+@Component
+class PersonBlockingRulesRepository {
+  @PersistenceContext
+  private val entityManager: EntityManager? = null
+
+  fun findMatchCandidates(person: Person, sourceSystem: String?, hasUUID: Boolean): List<PersonEntity> {
+    val query = entityManager!!.createNativeQuery(sql(person, sourceSystem, hasUUID), PersonEntity::class.java)
+    return query.resultList as List<PersonEntity>
+  }
+
+  fun sql(person: Person, sourceSystem: String?, hasPersonKey: Boolean): String {
+    var sql: String = ""
+    var sourceSystemCondition = when {
+      sourceSystem.isNullOrEmpty() -> "" else -> "AND pe1_0.source_system = '$sourceSystem'"
+    }
+    var personKeyCondition = when {
+      hasPersonKey -> "AND pe1_0.fk_person_key_id IS NOT NULL" else -> ""
+    }
+    person.getIdentifiersForMatching().forEach {
+      sql += """SELECT
+      pe1_0.id,pe1_0.birth_country,pe1_0.birth_place,pe1_0.crn,pe1_0.currently_managed,pe1_0.date_of_birth,pe1_0.defendant_id,pe1_0.ethnicity,pe1_0.first_name,pe1_0.last_name,pe1_0.master_defendant_id,pe1_0.merged_to,pe1_0.middle_names,pe1_0.nationality,pe1_0.fk_person_key_id,pe1_0.prison_number,pe1_0.religion,pe1_0.self_match_score,pe1_0.sex,pe1_0.sexual_orientation,pe1_0.source_system,pe1_0.title,pe1_0.version
```

Comment:
look into changing this into expression, since we already have alot of SQL.. makes the code mor readable